apps/explorer/lib/explorer/repo/config_helper.ex

```diff
@@ -29,10 +29,16 @@ defmodule Explorer.Repo.ConfigHelper do
     url = opts[:url] || System.get_env("DATABASE_URL")
     env_function = opts[:env_func] || (&System.get_env/1)
 
-    url
-    |> extract_parameters()
-    |> Keyword.merge(get_env_vars(@postgrex_env_vars, env_function))
-    |> Keyword.merge(get_env_vars(@app_env_vars, env_function))
+    if url == "" or is_nil(url) do
```

Comment:
If the url is nil we should fail and refuse to start the application, rather than create a default value that can't be used to connect to the database.

Reply:
@rkachowski I thought the idea of the ticket (https://app.zenhub.com/workspaces/data-services-sprint-60103a7bf450290020c3800c/issues/celo-org/data-services/161) was to make this particular env var optional?

Reply:
From what I understand, there are two ideas to the ticket:

1. Fix the bug reported by Viktor
2. Allow the application to connect to the database without providing `DATABASE_URL`

The line at https://github.com/celo-org/blockscout/blob/master/apps/explorer/lib/explorer/repo/config_helper.ex#L38 is already performing the check you have added.